iocore/aio/AIO.cc

```diff
@@ -514,9 +505,7 @@ aio_thread_main(void *arg)
       op->mutex     = op->action.mutex;
       if (op->thread == AIO_CALLBACK_THREAD_AIO) {
         SCOPED_MUTEX_LOCK(lock, op->mutex, thr_info->mutex->thread_holding);
-        if (!op->action.cancelled) {
-          op->action.continuation->handleEvent(AIO_EVENT_DONE, op);
-        }
+        op->handleEvent(EVENT_NONE, nullptr);
```

Comment:
Why do you remove this check ?

Reply:
Drop it ! You callback to io_complete instead.

Reply:
It will be checked within `io_complete`.